src/lightning_fabric/fabric.py

```diff
@@ -518,6 +521,44 @@ def launch(self, function: Optional[Callable[["Fabric"], Any]] = None, *args: An
             return self._strategy.launcher.launch(function, *args, **kwargs)
         return function(*args, **kwargs)
 
+    def call(self, hook_name: str, *args: Any, **kwargs: Any) -> None:
```

Comment:
what about also exposing fabric.on_batch_start? So a direct call? because this one again males it impossible to search for usage of a given function using your IDE.

I get that this one is more convenient, just wanted to raise the question:)

Reply:
I think we would have to do either one or the other. If we do both, it's one more thing we have to explain to the users what the difference is between both of them. 

Also, there is no callback interface, so even if you get the auto-suggestion through the type, there is no contract guaranteeing you that the hook exists on your callback (so you'd have to look it up anyway).

Reply:
cc @lantiga

Reply:
so the benefit is that the user could call whatever hook they implemented? That's sufficient reason for me not to go with the explicit version :)

Reply:
agreed, let's stay with strings for now

personally I find the naming doesn't evoke an event being emitted, rather `call` sounds like a direct call - so one wonders why you're not calling the function directly.

I'm not against it, but I would find something like `trigger` more intuitive.

this may just be me though, I'd like to hear from the others.

Reply:
Regarding the naming: I'd stick with `call` as this basically immediately calls the callbacks with some additional checks. `trigger` might indicate that we built a fully-fledged event-based system around this, which we don't want to do.

Reply:
At the moment, it is a direct call, just done through Fabric. The user could call the function directly too if they have the reference to the callback.

"call" also fits with **call**back

but I don't mind choosing trigger or emit if anybody prefers.

Reply:
@lantiga I agree we may need to iterate on the name, and get William Falcon's approval too. I suggest we merge the PR so we can also finish the examples, but keep it open to change the name from feedback.